src/Graph.ts

```diff
@@ -490,7 +521,15 @@ export default class Graph {
 			visit(curEntry);
 		}
 
-		return { orderedModules: ordered, dynamicImports, hasCycles };
+		// new items can be added during this loop
+		for (let i = 0; i < preservedModules.length; i++) {
+			curEntry = preservedModules[i];
+			curEntry.isEntryPoint = true;
+			curEntryHash = randomUint8Array(10);
+			visit(curEntry);
+		}
```

Comment:
It seems like this is done to ensure that no chunking happens right?

I think it can be done more easily by altering the visit function itself to do something like:

```js
if (!preserveModules)
  Uint8ArrayXor(module.entryPointsHash, curEntryHash);
else if (isZero(module.entryPointsHash))
  module.entryPointsHash = randomUint8Array(10);
```

then I think that would be all there is too it without any extra loops or need to track modules.

Reply:
Perfect! It got a lot simpler with this change.